src/views/Signin.jsx

```diff
@@ -0,0 +1,138 @@
+import React from 'react';
+import withFirebaseAuth from 'react-with-firebase-auth';
+import * as firebaseApp from 'firebase/app';
+import { useTranslation } from 'react-i18next';
+import 'firebase/auth';
+import {
+  Redirect,
+  Link,
+  useParams,
+  useLocation,
+} from 'react-router-dom';
+import Footer from '../components/Footer';
+import MailInput from '../components/MailInput';
+import fb from '../firebase';
+import { baseUrl } from '../appConfig';
+
+const firebaseAppAuth = firebaseApp.auth();
+
+const Signin = (props) => {
+  const [email, setEmail] = React.useState('');
+  const [password, setPassword] = React.useState('');
+  const [error, setError] = React.useState('');
+  const [passwordResetSuccess, setPasswordResetSuccess] = React.useState(false);
+  const location = useLocation();
+  const { t } = useTranslation();
+
+  const {
+    user,
+    signInWithEmailAndPassword,
+  } = props;
+  const { returnUrl } = useParams();
+
+  if (user) {
+    if (returnUrl) return <Redirect to={`/${decodeURIComponent(returnUrl)}`} />;
+    return user.emailVerified ? <Redirect to="/ask-for-help" /> : <Redirect to="/verify-email" />;
+  }
+
+  const reasonForSignin = location && location.state && location.state.reason_for_registration
+    ? location.state.reason_for_registration
+    : t('views.signIn.reasonForSigninDefault');
+  const headerText = t('views.signIn.headerText', { reasonForSignin });
+
+  const signIn = async (e) => {
+    e.preventDefault();
+    const signInResult = await signInWithEmailAndPassword(email, password);
+    if (signInResult.code) {
+      switch (signInResult.code) {
+        case 'auth/user-not-found': setError(t('views.signIn.noUser')); break;
+        case 'auth/wrong-password': setError(t('views.signIn.wrongUserOrPw')); break;
+        case 'auth/invalid-email': setError(t('views.signIn.emailInvalid')); break;
+        default: setError(signInResult.message);
```

Comment:
During my tests I watched the too-many-requests error and I thought it could be a good idea to catch that as well. 

```suggestion
      switch (signInResult.code) {
        case 'auth/user-not-found': setError(t('views.signIn.noUser')); break;
        case 'auth/wrong-password': setError(t('views.signIn.wrongUserOrPw')); break;
        case 'auth/invalid-email': setError(t('views.signIn.emailInvalid')); break;
        case 'auth/too-many-requests': setError(t('views.signIn.tooManyRequests')); break;
        default: setError(signInResult.message);
```